openmdao/core/group.py

```diff
@@ -1216,7 +1217,11 @@ def _setup_connections(self):
                             simple_warning(msg)
 
                 elif src_indices is not None:
-                    src_indices = np.atleast_1d(src_indices)
+                    if _is_slice(src_indices):
+                        val = self._abs_get_val(abs_out)
+                        src_indices = np.array([i for i in range(len(val[tuple(src_indices)]))])
```

Comment:
There are some problems here.  What if abs_out is a distributed variable?  In that case, src_indices will reference into the full distributed variable.  `val = self._abs_get_val(abs_out)` will only return the local part of the distributed variable.  Also, the value could have greater than a single dimension.  You really need a function that takes the global shape of the output and the 'fancy' indexing tuple and computes the index values along each dimension.  The current implementation will always give indices of [0, 1, 2, .....] which may not be correct.  This also means that we need a new test that has src_indices with distributed variables.